### PR TITLE
Some bug fixes related to bailout (in particular in planner)

### PR DESCRIPTION
**Pull request by atdotde**
### Describe the pull request:

- [x] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Fix the replanning of planned dives with bailout events. 

Show the dive mode in the planner notes (when changed)

Fix the planner notes with respect to set point changes.

Set the setpoint to 0 for non-CCR-segements in planner.

### Mentions:

Please test, I did some but found a number of bugs. In particular @willemferguson and @sfuchs79 .

### CODE REVIEW

File: qt-models/diveplannermodel.cpp [hunk 1]
```diff
@@ -332,8 +337,13 @@ bool DivePlannerPointsModel::setData(const QModelIndex &index, const QVariant &v
 			CylindersModel::instance()->updateTrashIcon();
 			break;
 		case DIVEMODE:
-			if (value.toInt() < FREEDIVE) // FIXME: I want to be a combo box and translate strings to enum values
+			if (value.toInt() < FREEDIVE) {
 				p.divemode = (enum divemode_t) value.toInt();
+				if (p.divemode == CCR)
+					p.setpoint = prefs.defaultsetpoint;
+				else
+					p.setpoint = 0;
```
Comment (neolit123) [inline qt-models/diveplannermodel.cpp]: probably best to one-line this:
```
p.setpoint = p.divemode == CCR ? prefs.defaultsetpoint : 0;
```

**Review by neolit123 — COMMENTED** [review]
added one comment + restarted travis.

**Comment by dirkhh** [comment]
I just look at you for this code @atdotde - please request reviews from other people as you see fit.

**Comment by atdotde** [comment]
¯\_(ツ)_/¯

**Comment by sfuchs79** [comment]
I tested this and can confirm that the one bug I reported on the ML (dive modes not loaded on replanning) is fixed.

**Comment by atdotde** [comment]
Nobody complained, so let's get this in the next release.

**Pull request merged**